From 0c07c704712218b915d55cd22092f9d3fc52c9eb Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@st.com>
Date: Fri, 29 Oct 2021 11:36:15 +0200
Subject: [PATCH 022/954] feat(stm32mp1): add Backup SRAM definitions

Backup SRAM is an embedded RAM that is kept on during low-power phases.
Its address is 0x54000000. Its size is 8kB on STM32MP15, and 4kB on
STM32MP13.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 9989cbd2e2c34edb3259c19028f8ba352586b395 feat(stm32mp1): add Backup SRAM definitions]
Change-Id: Ia863d4100df211f5356df5f0d6bd1d0776d25d27
Signed-off-by: Yann Gautier <yann.gautier@st.com>
---
 plat/st/stm32mp1/stm32mp1_def.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/plat/st/stm32mp1/stm32mp1_def.h b/plat/st/stm32mp1/stm32mp1_def.h
index 0529c787c6..777288b50f 100644
--- a/plat/st/stm32mp1/stm32mp1_def.h
+++ b/plat/st/stm32mp1/stm32mp1_def.h
@@ -107,6 +107,14 @@
 #define STM32MP_SYSRAM_SIZE		U(0x00040000)
 #endif /* STM32MP15 */
 
+#define STM32MP_BACKUP_RAM_BASE		U(0x54000000)
+#if STM32MP13
+#define STM32MP_BACKUP_RAM_SIZE		U(0x00002000)	/* 8KB */
+#endif /* STM32MP13 */
+#if STM32MP15
+#define STM32MP_BACKUP_RAM_SIZE		U(0x00001000)	/* 4KB */
+#endif /* STM32MP15 */
+
 #define STM32MP_NS_SYSRAM_SIZE		PAGE_SIZE
 #define STM32MP_NS_SYSRAM_BASE		(STM32MP_SYSRAM_BASE + \
 					 STM32MP_SYSRAM_SIZE - \
-- 
2.53.0

